packages/integrations/src/captureconsole.ts

```diff
@@ -1,6 +1,23 @@
 import type { EventProcessor, Hub, Integration } from '@sentry/types';
+import type { ConsoleLevel } from '@sentry/utils';
 import { CONSOLE_LEVELS, fill, GLOBAL_OBJ, safeJoin, severityLevelFromString } from '@sentry/utils';
 
+interface CaptureConsoleOptions {
+  /**
+   * The console levels to be captured.
+   */
+  levels: Array<ConsoleLevel>;
+  /**
+   * A callback function that determines if console call should be captured as event.
+   *
+   * @param args Arguments
+   * @returns A boolean indicating whether the call should be captured.
+   */
+  beforeCapture(level: ConsoleLevel, ...args: unknown[]): boolean;
```

Comment:
m: Not 100% sure, but should we make this return `boolean | void`, and only check on `=== false`? 🤔 but maybe it is better to be "stricter" 🤔